### ISSUE TITLE
Better logging around GraphQL API usage

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#5706]
Right now, it is possible to construct very heavy GraphQL API queries that harm Sourcegraph (I have opened a separate issue for us to prevent this, see https://github.com/sourcegraph/sourcegraph/issues/5705 ) and we currently do not have any way to identify exactly which queries those were. Today we log:

1. That a GraphQL query was made (but no information about it), but only after it completed.
2. How long the GraphQL query took, but only after it completed.

With such queries, the above log lines do not even print, so it is impossible to identify what queries might have caused an issue. We should be logging:

1. The authenticated user, if any.
2. The GraphQL query string (without this, I don't see how we could debug such situations).

**Comment by dadlerj** [comment]
This is relevant to the concept of a management audit log (e.g. who viewed which repos at what times, who edited what settings at what times, etc.) in addition to the metrics and monitoring side of things.

Not sure what you had in mind here, but this might make more sense from a product POV in an admin UI than in backend logs.

**Comment by slimsag** [comment]
Having this as part of auditing also makes sense, for sure. But that should be tracked separately. The goal of this issue is to have the most basic form of debugging for when a customer hits into the problem described in https://github.com/sourcegraph/sourcegraph/issues/5705 and their instance is suffering a major outage.

I may raise the priority on this and ship it in a patch soon, depending on what I observe in some customer deployments.

**Comment by tsenart** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.9 release is scheduled for tomorrow** at 10:00 CEST.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by tsenart** [comment]
Moving this to Backlog since it doesn't look like it got done in time for 3.9. Please update accordingly otherwise.

**Comment by unknwon** [comment]
^ I'm moving this to 3.10 milestone based on above event.

**Comment by tsenart** [comment]
@unknwon: Which event?

**Comment by slimsag** [comment]
Thanks, yes, I plan to do this in 3.10 not 3.9.1.

**Comment by unknwon** [comment]
Oh, I was referring to this one:

![Screen Shot 2019-11-03 at 10 26 43 AM](https://user-images.githubusercontent.com/2946214/68090065-6f3c2080-fe24-11e9-9de4-fc550e6aebc9.png)

**Comment by slimsag** [comment]
TODO as part of this issue: Stephen to axe / figure out what to do with `src_graphql_field_seconds` histogram metric

**Comment by beyang** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.10 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by beyang** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.11 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you
